### ISSUE TITLE
TypeError: Cannot read property 'run' of undefined at examples/window.js

### ISSUE THREAD

**Issue by Diegox777** [nodefluent/kafka-streams#105]
`TypeError: Cannot read property 'run' of undefined
    at Merge.run (/home/root/projects/dev/kafka-streams/node_modules/most/lib/combinator/merge.js:95:40)
    at Slice.run (/home/root/projects/dev/kafka-streams/node_modules/most/lib/combinator/slice.js:94:40)
    at Tap.run (/home/root/projects/dev/kafka-streams/node_modules/most/lib/combinator/transform.js:67:22)
    at runSource (/home/root/projects/dev/kafka-streams/node_modules/most/lib/runSource.js:39:35)
    at /home/root/projects/dev/kafka-streams/node_modules/most/lib/runSource.js:31:5
    at new Promise (<anonymous>)
    at withScheduler (/home/root/projects/dev/kafka-streams/node_modules/most/lib/runSource.js:30:10)
    at withDefaultScheduler (/home/root/projects/dev/kafka-streams/node_modules/most/lib/runSource.js:26:10)
    at drain (/home/root/projects/dev/kafka-streams/node_modules/most/lib/combinator/observe.js:36:46)
    at observe (/home/root/projects/dev/kafka-streams/node_modules/most/lib/combinator/observe.js:26:10)
    at Stream._Stream2.default.observe._Stream2.default.forEach (/home/root/projects/dev/kafka-streams/node_modules/most/lib/index.js:209:31)
    at KStream.forEach (/home/root/projects/dev/kafka-streams/lib/dsl/StreamDSL.js:271:29)
    at Object.<anonymous> (/home/root/projects/dev/kafka-streams/examples/window.js:20:6)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
(node:9247) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:9247) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:9247) UnhandledPromiseRejectionWarning: TypeError: window.getStream(...).error is not a function
    at stream$.skipWhile.takeWhile.until.tap.drain.then.catch.e (/home/root/projects/dev/kafka-streams/lib/dsl/KStream.js:321:44)
    at <anonymous>
    at runMicrotasksCallback (internal/process/next_tick.js:121:5)
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
    at Function.Module.runMain (module.js:695:11)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3`

**Comment by chitkosarvesh** [comment]
I am having the same problem too, even with the latest release of the library. Can someone please help?
Thanks

**Comment by KlajdiDosti** [comment]
I am having the same problem too, 
```
TypeError: Cannot read property 'run' of undefined
    at new UpperBound (/var/www/alpha-service/authentication-service/node_modules/most/lib/combinator/timeslice.js:117:28)
    at Until.run (/var/www/alpha-service/authentication-service/node_modules/most/lib/combinator/timeslice.js:51:13)
    at Tap.run (/var/www/alpha-service/authentication-service/node_modules/most/lib/combinator/transform.js:67:22)
    at runSource (/var/www/alpha-service/authentication-service/node_modules/most/lib/runSource.js:39:35)
    at /var/www/alpha-service/authentication-service/node_modules/most/lib/runSource.js:31:5
    at new Promise (<anonymous>)
    at withScheduler (/var/www/alpha-service/authentication-service/node_modules/most/lib/runSource.js:30:10)
    at withDefaultScheduler (/var/www/alpha-service/authentication-service/node_modules/most/lib/runSource.js:26:10)
    at drain (/var/www/alpha-service/authentication-service/node_modules/most/lib/combinator/observe.js:36:46)
    at Stream._Stream2.default.drain (/var/www/alpha-service/authentication-service/node_modules/most/lib/index.js:221:29)
    at KStream.window (/var/www/alpha-service/authentication-service/node_modules/kafka-streams/lib/dsl/KStream.js:326:14)
    at RolesService.startRolesStreamsValPass (/var/www/alpha-service/authentication-service/services/roles.js:162:31)
    at RolesService.start (/var/www/alpha-service/authentication-service/services/roles.js:36:9)
    at new RolesService (/var/www/alpha-service/authentication-service/services/roles.js:29:8)
    at Object.<anonymous> (/var/www/alpha-service/authentication-service/services/roles.js:201:18)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/var/www/alpha-service/authentication-service/api/controllers/roleController.js:1:84)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/var/www/alpha-service/authentication-service/api/routes/role.js:3:22)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (/var/www/alpha-service/authentication-service/api/routes/index.js:8:18)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
```

**Comment by krystianity** [comment]
@Diegox777  @chitkosarvesh fixed in kafka-streams@4.9.0 sorry for the frustration.

**Comment by chitkosarvesh** [comment]
Thanks!
